logger/clickhouse.cc

```diff
@@ -94,6 +95,52 @@ TfwClickhouse::make_block()
 	block_.Reserve(max_events_ * 2);
 }
 
+bool
+TfwClickhouse::do_reconnect() {
+	try {
+		client_.reset();
+
+		client_ = std::make_unique<ch::Client>(client_options_);
+
+		spdlog::info("Successfully reconnected to ClickHouse");
+
+		needs_reconnect.store(false);
+		last_reconnect_attempt.store(std::chrono::steady_clock::now());
+		return true;
+	}
+	catch (const std::exception& e) {
+		spdlog::error("Failed to reconnect to ClickHouse: {}", e.what());
+		last_reconnect_attempt.store(std::chrono::steady_clock::now());
+		return false;
+	}
+	catch (...) {
+		spdlog::error("Failed to reconnect to ClickHouse: unknown error");
+		last_reconnect_attempt.store(std::chrono::steady_clock::now());
+		return false;
+	}
+}
+
+void
+TfwClickhouse::append_timestamp(uint64_t timestamp)
+{
+	block_[0]->As<ch::ColumnDateTime64>()->Append(timestamp);
+}
+
+template <typename ColType, typename ValType>
+void
+TfwClickhouse::append_int(TfwBinLogFields field, ValType value)
+{
+	const size_t col_index = field_to_column_index(field);
```

Comment:
I think this can be constexpr since we pass `field` from the very upper level as a constant